### PR TITLE
Topic/ttonchevvmware/defined entities doc

### PR DESCRIPTION
**Pull request by ttonchevvmware**

### CODE REVIEW

File: documentation/images/rde_lifecycle.png
[the remark this answers is not in the document]
Comment (ttonchevvmware) [reply documentation/images/rde_lifecycle.png:1]: Will add a comment about that. The diagram is getting too busy.

File: documentation/extensibility-platform/defined-entities/rde-hooks.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+}
+```
+
+The type definition binds three of the interface behaviors to different lifecycle events of the entities of the type:
+
+- the `autoResolve` behavior will be invoked after the initial entity creation
+- the `processUpdate` behavior will be invoked every time the entity is modified
+- the `validateDelete` behavior will be invoked to check whether the entity can be deleted or not
+
+## Available RDE Lifecycle Hooks
```
[the remark this answers is not in the document]
Comment (ttonchevvmware) [reply documentation/extensibility-platform/defined-entities/rde-hooks.md:48]: I am adding an additional paragraph discussing that.

**Review by theotonchev — APPROVED** [review]

File: documentation/extensibility-platform/defined-entities/rde-hooks.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (theotonchev) [reply documentation/extensibility-platform/defined-entities/rde-hooks.md:48]: See #Hook Execution Control

File: documentation/images/rde_lifecycle.png
[the remark this answers is not in the document]
Comment (theotonchev) [reply documentation/images/rde_lifecycle.png:1]: It appears the comment was lost. Adding it again.

File: documentation/extensibility-platform/defined-entities/rde-hooks.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+automatically and the entity state will be switched to `RESOLVED` if the content validation
+against the schema is successful.
+
+If the behavior execution fails, the entity state will switched to the `RESOLUTION_ERROR` state.
+
+### PostUpdate
+
+A behavior bound to the `PostUpdate` hook will be invoked immediately after an entity
+of the type is modified. The behavior may perform operations like allocating external resources.
+It can also update the entity contents.
```
[the remark this answers is not in the document]
Comment (theotonchev) [reply documentation/extensibility-platform/defined-entities/rde-hooks.md:66]: No, only on PostCreate. Any needed resolution has to be manual.

**Pull request merged**